intro-to-R-tidyverse/02-intro_to_tidyverse.Rmd

```diff
@@ -289,9 +310,68 @@ Let's use `head(metadata_summary)` so we can look at a preview of what we made:
 head(metadata_summary)
 ```
 
+Let's say we have a scenario where we have two data.frames that we would like to 
+combine. 
+Recall that `metadata_pipe` and `metadata_filtered` are data.frames we've
+made with information about some of the same samples.
+The [`dplyr::join` family of functions](https://dplyr.tidyverse.org/reference/join.html) 
+are useful for various scenarios of combining data.frames. 
+
+For now, we will focus on `inner_join`, which will combine data.frames by only
+keeping information about the samples that are in both data.frames. 
+We need to use the `by` argument to designate what column(s) 
+should be used as a key to match the data.frames (in this case our `sample_id`s). 
+
+```{r}
+metadata_filtered %>% inner_join(metadata_pipe, by = 'sample_id') 
+```
+
+Note that if there are columns of the same name in both data.frames, the default 
+is to save both columns but a `x` or `y` is added to the name to note which 
+data.frame it originated from.
+In this case, `x` = `metadata_filtered` and `y` = `metadata_pipe`.
+
+## A brief intro to `ggplot2` data visualization
+
+`ggplot2` is a tidyverse package that is very useful for making publication ready 
+plots. 
+We will use `ggplot2` to create data visualization throughout our workshops. 
+For now, we will introduce you to the very basic structure of `ggplot2`. 
+
+The two basic arguments needed to start a ggplot are 1) `data` and 2) `aes` (short for
+aesthetics).
+Other aspects of the plot will be added after these two arguments are established. 
+
+For the `data` argument, `ggplot` needs a data.frame that is ["tidy"](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html)
+
+For the `aes` argument, we need to specify our variables. 
+The variables we specify should be named columns within the object we specified as 
+our `data`. 
+Here we will use `sex` and `age_mths` as our variables. 
+
+```{r}
+sex_by_age_plot <- ggplot(data = metadata, aes(x = sex, y = age_mths))
```

Comment:
This plot will not show up without adding `sex_by_age_plot` on it's own line or in it's own chunk, is that correct? If we're going to say 
> Although we have the basic structure of our ggplot, we haven't yet added anything to specify what kind of plot we would like, so this default plot is rather boring and not sensible. 

below, we should make sure we're displaying the plot in the notebook.

Reply:
Ah. True.